spec/features/dataset_show_page_spec.rb

```diff
@@ -199,24 +199,45 @@
       expect(page).to have_content('Topic: Not added')
     end
 
-    context 'When a dataset has no datafiles' do
-      it 'Last Updated field displays last_updated_at' do
+    context 'When public_updated_at is present on a dataset' do
+      scenario 'Last Updated field displays public_updated_at' do
         dataset = DatasetBuilder.new.build
         index_and_visit(dataset)
         expect(page).to have_content("Last updated: #{dataset['last_updated_at']}")
       end
     end
 
-    context 'When a dataset has datafiles' do
-      it 'Last Updated field is the most recent datafiles updated_at date' do
+    context 'When public_updated_at is not present on a dataset with no datafiles' do
+      scenario 'Last Updated field displays dataset last_updated_at' do
+        dataset = DatasetBuilder.new.build
+        dataset.delete(:public_updated_at)
+        index_and_visit(dataset)
+        expect(page).to have_content("Last updated: #{dataset['last_updated_at']}")
+      end
+    end
+
+    context 'When public_updated_at is not present on a dataset with datafiles' do
+      scenario 'Last Updated field displays most recent datafile updated_at' do
+        datafile = [
```

Comment:
Can you reuse the existing datafile constants from the dataset_spec_builder?

Reply:
Also, datafile is technically datafileS - maybe it would be shorter to just have it as a hash and then to `.with_datafiles([datafile])`.